catalyst/utils/distributed.py

```diff
@@ -124,17 +125,20 @@ def get_slurm_params():
     current_node = os.environ["SLURMD_NODENAME"]
     master_node = socket.gethostbyname(nodes[0])
     cur_node_idx = nodes.index(current_node)
-    return cur_node_idx, num_nodes, master_node
+    job_id = os.getenv("SLURM_JOB_ID")
+    master_port = str(5*10**4 + int(job_id) % 10**4)
```

Comment:
could we first try to take `master_port ` for os.env and randomly generate it only for None case?

Reply:
done

Reply:
not sure if :)

Reply:
We call this function if slurm available. By default, SLURM_JOB_ID must be defined. If SLURM_JOB_ID is None, it seems something wrong with slurm, and raising an exception is a good option.